mkosi/__init__.py

```diff
@@ -4170,45 +4162,59 @@ def install_unified_kernel(
     if do_run_build_script:
         return
 
+    prefix = "boot" if args.get_partition(PartitionIdentifier.xbootldr) else "efi"
+
     with mount(), complete_step("Generating combined kernel + initrd boot file…"):
-        # Apparently openmandriva hasn't yet completed its usrmerge so we use lib here instead of usr/lib.
-        with os.scandir(root / "lib/modules") as d:
-            for kver in d:
-                if not (kver.is_dir() and os.path.isfile(os.path.join(kver, "modules.dep"))): # type: ignore
-                    continue
+        for kver, kimg in gen_kernel_images(args, root):
+            if args.image_id:
+                image_id = args.image_id
+                if args.image_version:
+                    partlabel = f"{args.image_id}_{args.image_version}"
+                else:
+                    partlabel = f"{args.image_id}"
+            else:
+                image_id = "linux"
```

Comment:
That fallback is maybe a bit generic. How about `mkosi` or `mkosi-$DISTRO`? (I know it's lifted from the current script, but I was thinking about the docu for `IMAGE_ID` in the yet unreleased version of the `/etc/os-release` man page)